### PR TITLE
test: enable browser tests

### PR DESCRIPTION
**Pull request by antfu**
### 🔗 Linked issue

https://github.com/nuxt/framework/issues/3198

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: test/utils.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  page.on('console', (message) => {
+    consoleLogs.push({
+      type: message.type(),
+      text: message.text()
+    })
+  })
+  page.on('pageerror', (err) => {
+    pageErrors.push(err)
+  })
```
Comment (antfu) [inline test/utils.ts:18]: Maybe this is general enough to be included in the utils directly?

Comment (pi0) [reply test/utils.ts:18]: I'm okay by adding utils like this directly to test-utils.

File: test/utils.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  if (path) {
+    await page.goto(url(path), { waitUntil: 'networkidle' })
+  }
+
+  const consoleLogErrors = consoleLogs.filter(i => i.type === 'error')
```
[the remark this answers is not in the document]
Comment (antfu) [reply test/utils.ts]: I'd like to reuse `createPage`
https://github.com/nuxt/framework/blob/66605971a887916428a1abba056b872408a1f1ef/packages/test-utils/src/browser.ts#L37-L46

However, the problem is that the listeners have to be called before `page.goto`. We might have an option `spy` to toggle if the listeners are needed, or ever have some composable hooks like `onPageCreated` `onPageRendered` etc. Plus we need also pass options for `page.goto` for `waitUntil: 'networkidle'`

[the remark this answers is not in the document]
Comment (pi0) [reply test/utils.ts]: Maybe support a  callback option in `BrowserContextOptions` in order to hook before navigation?

[the remark this answers is not in the document]
Comment (antfu) [reply test/utils.ts]: Guess we could merge this for now and think to include it in test-utils later? As we improve the test for different parts, we might learn more potential usage about it

**Pull request merged**